comm.go

```diff
@@ -74,21 +77,16 @@ func (p *PubSub) handleSendingMessages(ctx context.Context, s inet.Stream, outgo
 			if !ok {
 				return
 			}
-			if dead {
```

Comment:
Shouldn't need this anymore (I believe this existed as a hack to prevent some deadlocks but those are no longer possible).

Reply:
I believe you are correct. This was because sends on the other end of this outgoing channel could get filled up and block the loops. All those sends are now coupled with default selects (though please double check this for me).